bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -161,7 +164,7 @@ public SingleDirectoryDbLedgerStorage(ServerConfiguration conf, LedgerManager le
         this.writeCacheMaxSize = writeCacheSize;
         this.writeCache = new WriteCache(allocator, writeCacheMaxSize / 2);
         this.writeCacheBeingFlushed = new WriteCache(allocator, writeCacheMaxSize / 2);
-
+        this.writeCacheLastFlushed = new WriteCache(allocator, writeCacheMaxSize / 2);
```

Comment:
This will make us use 150% of the configured memory for the write cache.

Reply:
@merlimat 
writeCacheBeingFlushe and writeCacheLastFlushed almost  only one of the two exists at the same time
so I think this just 100% memory for the write cache.
<img width="1420" alt="image" src="https://user-images.githubusercontent.com/42990025/174243987-33261c58-0289-47f3-af5d-c7b0768228b3.png">

Reply:
The "flush" to disk is not instantaneous. In fact, it can take a very long time and during that time all 3 segments will be filled up during that time.

Reply:
> 

@merlimat 
thanks for your suggestion，I have update the init code
<img width="1248" alt="image" src="https://user-images.githubusercontent.com/42990025/174589904-f7900999-cc3e-4921-9618-5ba182f9596a.png">